test/integration/hooks/validate.test.js

```diff
@@ -41,6 +41,74 @@ describe(Support.getTestDialectTeaser('Hooks'), function() {
       });
     });
 
+    describe('#3534, hooks modifications', function() {
+      it('fields modified in hooks are saved', function() {
+        var self = this;
+
+        this.User.afterValidate(function(user, options) {
+          //if username is defined and has more than 5 char
+          user.username = user.username
+                          ? (user.username.length < 5 ? null : user.username)
+                          : null;
+          user.username = user.username || 'Samorost 3';
+
+        });
+
+        this.User.beforeValidate(function(user, options) {
+          user.mood = user.mood || 'neutral';
+        });
+
+
+        return this.User.create({username: 'T', mood: 'neutral'}).then(function(user) {
+          expect(user.mood).to.equal('neutral');
+          expect(user.username).to.equal('Samorost 3');
+
+          //change attributes
+          user.mood = 'sad';
+          user.username = 'Samorost Good One';
+
+          return user.save();
+        }).then(function(uSaved) {
+          expect(uSaved.mood).to.equal('sad');
+          expect(uSaved.username).to.equal('Samorost Good One');
+
+          //change attributes, expect to be replaced by hooks
+          uSaved.username = 'One';
+
+          return uSaved.save();
+        }).then(function(uSaved) {
+          //attributes were replaced by hooks ?
+          expect(uSaved.mood).to.equal('sad');
+          expect(uSaved.username).to.equal('Samorost 3');
+          return self.User.findById(uSaved.id);
```

Comment:
But we have no test to check this while hooks are changing the values. I am looking for any cases that may involve 

- Creating
- Finding
- Updating

because they all pass from different methods and hooks, we need to check if there is any issue in these methods when hooks modification are in effect